jetbrains-core/src/software/aws/toolkits/jetbrains/utils/SpinUtils.kt

```diff
@@ -0,0 +1,19 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.utils
+
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import java.time.Duration
+
+fun spinUntil(duration: Duration, condition: () -> Boolean) {
+    val start = System.nanoTime()
+    runBlocking {
+        while (!condition()) {
+            if (System.nanoTime() - start > duration.toNanos())
+                throw IllegalStateException("Condition not reached within $duration")
+            delay(1)
+        }
```

Comment:
Just for my own enlightenment: `runBlocking` blocks the current thread, which defeats the typical purpose of a coroutine. What's the advantage of `runBlocking{ delay(1) }` vs ye olde `Thread.sleep` ?

Reply:
In this case it is the same, since it uses the callee thread for the coroutine